libs/server/Servers/GarnetServerOptions.cs

```diff
@@ -284,6 +284,11 @@ public class GarnetServerOptions : ServerOptions
         /// </summary>
         public int ReplicaSyncDelayMs = 5;
 
+        /// <summary>
+        /// Upper bound on the lag (i.e. throttle replicaAOF append if AOF.TailAddress - ReplicationOffset > ReplicaMaxLag) between primary and replica. -1 - Synchronous replay, >= 0 - background replay with specified lag 
+        /// </summary>
+        public int ReplicaMaxLag = 32768;
```

Comment:
this default looks too small and will definitely cause backup at primary every time a checkpoint is taken at the replica. why not set this to -1 as default, i.e. we do not gate the primary due to a slow replica.

Reply:
Good point! This way we make async replay opt in.

Reply:
I think infinite lag should be the default. That's the right way to run the system instead of lock step.

Reply:
Ok, made the following change:

0: Synchronous replay, >=1: background replay with specified lag, -1: infinite lag